### PR TITLE
Add feature settings, user optIns, and optIn filter to types

### PR DESCRIPTION
**Pull request by emmawillis**

### CODE REVIEW

File: lib/shared/bucketing-assembly-script/assembly/types/configBody.ts [hunk 1]
```diff
@@ -12,24 +12,22 @@ export class PublicProject extends JSON.Value {
...
-    settings: JSON.Obj | null
+    settings: JSON.Obj
 
     constructor(project: JSON.Obj) {
         super()
         this._id = getStringFromJSON(project, '_id')
         this.key = getStringFromJSON(project, 'key')
         this.a0_organization = getStringFromJSON(project, 'a0_organization')
-        this.settings = getJSONObjFromJSONOptional(project, 'settings')
+        this.settings = getJSONObjFromJSON(project, 'settings')
```
Comment (ajwootto) [inline lib/shared/bucketing-assembly-script/assembly/types/configBody.ts:22]: does every config in the CDN have a settings field now?

File: lib/shared/bucketing-assembly-script/assembly/types/dvcUser.ts [hunk 2]
```diff
@@ -26,6 +26,7 @@ export class DVCUser extends JSON.Obj implements DVCUserInterface {
     appVersion: string | null
     appBuild: f64
     customData: JSON.Obj | null
+    optIns: JSON.Obj | null
```
Comment (ajwootto) [inline lib/shared/bucketing-assembly-script/assembly/types/dvcUser.ts:29]: this isn't something that can be passed in is it? If not it shouldn't be on DVCUser

Comment (emmawillis) [reply lib/shared/bucketing-assembly-script/assembly/types/dvcUser.ts:29]: yeah but that would require an sdk update for all the sdks that use this. the idea was that the optIn data is stored in edgeDB as user.optIns, so all the places where we combine edgeDB data with user data would automatically handle it by adding it to the DVCUser object that gets sent for bucketing, without needing an update

File: lib/shared/bucketing-assembly-script/assembly/types/configBody.ts (hunk 1 above)
Comment (emmawillis) [reply lib/shared/bucketing-assembly-script/assembly/types/configBody.ts:22]: https://github.com/DevCycleHQ/js-sdks/blob/main/lib/shared/types/src/types/config/models/project.ts#L12 we made it required here already and treat it as required in other places, so I think so

File: lib/shared/types/src/types/apis/sdk/clientSDKAPI.ts [hunk 3]
```diff
@@ -116,6 +116,13 @@ export class DVCAPIUser {
     @Transform(({ value }) => isString(value) ? JSON.parse(value) : value)
         privateCustomData?: DVCJSON
 
+    /**
+     * JSON data recording user opt-in features.
+     */
+     @IsOptional()
+     @Transform(({ value }) => isString(value) ? JSON.parse(value) : value)
+         optIns?: Record<string, boolean>
```
Comment (ajwootto) [inline lib/shared/types/src/types/apis/sdk/clientSDKAPI.ts:124]: this can't be passed to the API can it?

**Review by kaushalkapasi — APPROVED** [review]

**Pull request merged**